src/DesignQuery/DesignQuery.cpp

```diff
@@ -57,7 +57,121 @@ extern FOEDAG::Session* GlobalSession;
 using namespace FOEDAG;
 using Time = std::chrono::high_resolution_clock;
 using ms = std::chrono::milliseconds;
+using json = nlohmann::ordered_json;
+
+std::filesystem::path DesignQuery::GetProjDir() const {
+  ProjectManager* projManager = m_compiler->ProjManager();
+  std::filesystem::path dir(projManager->getProjectPath().toStdString());
+  return dir;
+}
+
+std::filesystem::path DesignQuery::GetHierInfoPath() const {
+  std::filesystem::path dir = GetProjDir();
+  std::filesystem::path hier_info = "hier_info.json";
+  return dir / hier_info;
+}
+
+std::filesystem::path DesignQuery::GetPortInfoPath() const {
+  std::filesystem::path dir = GetProjDir();
+  std::filesystem::path port_info = "port_info.json";
+  return dir / port_info;
+}
 
 bool DesignQuery::RegisterCommands(TclInterpreter* interp, bool batchMode) {
+  auto parse_design_data = [](void* clientData, Tcl_Interp* interp, int argc,
+                              const char* argv[]) -> int {
+    DesignQuery* design_query = (DesignQuery*)clientData;
+    Compiler* compiler = design_query->GetCompiler();
+    bool status = true;
+
+    json& hier_json = design_query->getHierJson();
+    json& port_json = design_query->getPortJson();
+
+    std::filesystem::path port_info_path = design_query->GetPortInfoPath();
+    std::filesystem::path hier_info_path = design_query->GetHierInfoPath();
+    if (!FileUtils::FileExists(port_info_path)) {
+      status = false;
+      compiler->Message("port_info.json not found");
```

Comment:
I think we should print the path too.
We will be adding hyperlinks to paths in log messages so it will be helpful then.

Reply:
Will do!